File: Go/fibonacci_series.go
// Fibonacci series

package main

import (
	"fmt"
	"strconv"
)

// Fibonacci series using loops.
func FibonacciLoop(n int) int {
	f := make([]int, n+1, n+2)
	if n < 2 {
		f = f[0:2]
	}
	f[0] = 0
	f[1] = 1
	for i := 2; i <= n; i++ {
		f[i] = f[i-1] + f[i-2]
	}
	return f[n]
}

// Fibonacci series using recursive function.
func FibonacciRecursion(n int) int {
	if n <= 1 {
		return n
	}
	return FibonacciRecursion(n-1) + FibonacciRecursion(n-2)
}

func main() {
	for i := 0; i <= 9; i++ {
		fmt.Print(strconv.Itoa(FibonacciLoop(i)) + " ")
	}
	fmt.Println("")
	for i := 0; i <= 9; i++ {
		fmt.Print(strconv.Itoa(FibonacciRecursion(i)) + " ")
	}
	fmt.Println("")
}

File: Java/FibonacciSeries.java
/* This program uses a recursive function to find the number present at the given position in the Fibonacci series */

import java.util.*;

public class FibonacciSeries {
   public static long fib(long n) {
      if ((n == 0) || (n == 1))
         return n;
      else
         return fib(n - 1) + fib(n - 2);
   }
   public static void main(String[] args)
   {
      Scanner sc=new Scanner(System.in);
      System.out.println("Please Enter the Position:");
      int pos=sc.nextInt();
      if (pos == 1)
            System.out.println("The " + pos + "st fibonacci number is: " + fib(pos));
      if (pos == 2)
            System.out.println("The " + pos + "nd fibonacci number is: " + fib(pos));
      if (pos == 3)
            System.out.println("The " + pos + "rd fibonacci number is: " + fib(pos));
      else     
            System.out.println("The " + pos + "th fibonacci number is: " + fib(pos));
   }
}
